pallets/staking/src/inflation.rs

```diff
@@ -34,17 +34,275 @@ pub fn compute_total_payout<N>(
     npos_token_staked: N,
     total_tokens: N,
     era_duration: u64,
+    max_inflated_issuance: N,
+    non_inflated_yearly_reward: N,
 ) -> (N, N)
 where
     N: AtLeast32BitUnsigned + Clone,
 {
     // Milliseconds per year for the Julian year (365.25 days).
     const MILLISECONDS_PER_YEAR: u64 = 1000 * 3600 * 24 * 36525 / 100;
-
     let portion = Perbill::from_rational_approximation(era_duration as u64, MILLISECONDS_PER_YEAR);
-    let payout = portion
-        * yearly_inflation
-            .calculate_for_fraction_times_denominator(npos_token_staked, total_tokens.clone());
-    let maximum = portion * (yearly_inflation.maximum * total_tokens);
-    (payout, maximum)
+    // Check whether the fixed rewards kicks in?.
+    if total_tokens >= max_inflated_issuance {
+        let payout = portion * non_inflated_yearly_reward;
+        // payout is always maximum.
+        (payout.clone(), payout)
+    } else {
+        let payout = portion
+            * yearly_inflation
+                .calculate_for_fraction_times_denominator(npos_token_staked, total_tokens.clone());
+        let maximum = portion * (yearly_inflation.maximum * total_tokens);
+        (payout, maximum)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use sp_runtime::{curve::PiecewiseLinear, Perbill};
+
+    pallet_staking_reward_curve::build! {
+        const I_NPOS: PiecewiseLinear<'static> = curve!(
+            min_inflation: 0_025_000,
+            max_inflation: 0_100_000,
+            ideal_stake: 0_500_000,
+            falloff: 0_050_000,
+            max_piece_count: 40,
+            test_precision: 0_005_000,
+        );
+    }
+
+    #[test]
+    fn npos_curve_is_sensible() {
+        const YEAR: u64 = 365 * 24 * 60 * 60 * 1000;
+        const MAX_INFLATED_TOTAL_ISSUANCE: u64 = 1_000_000_000;
+        const NON_INFLATED_TOTAL_YEARLY_REWARD: u64 = 1_000_000;
+
+        // check maximum inflation.
+        // not 10_000 due to rounding error.
+        assert_eq!(
+            super::compute_total_payout(
+                &I_NPOS,
+                0,
+                100_000u64,
+                YEAR,
+                MAX_INFLATED_TOTAL_ISSUANCE,
+                NON_INFLATED_TOTAL_YEARLY_REWARD
```

Comment:
Additionally, a closure could be made from `compute_total_payout` which provides these two arguments and takes the rest as parameters.